### ISSUE TITLE
Openlayers views don't support non-specification of basemap or stamen type as suggested by ckanext-spatial notes on common basemap

### ISSUE THREAD

**Issue by mattfullerton** [ckan/ckanext-geoview#57]
In the docs, we're told that geoview supports the same options as the ckanext-spatial map widgets. However, specifying type = stamen does not work because it isn't accounted for in this code:
https://github.com/ckan/ckanext-ghttps://github.com/ckan/ckanext-geoview/blob/master/ckanext/geoview/public/js/ol_preview.js#L113

Also, the handler for no setting (https://github.com/ckan/ckanext-geoview/blob/master/ckanext/geoview/public/js/ol_preview.js#L129) doesn't work because OL_HELPERS.createLayerFromConfig (https://github.com/ckan/ckanext-geoview/blob/master/ckanext/geoview/public/js/ol_preview.js#L138) expects the type to be set:
https://github.com/ckan/ckanext-geoview/blob/master/ckanext/geoview/public/js/vendor/ol-helpers/ol-helpers.js#L1473

...the setting ought to be "XYZ", which is easy to fix, but this leads to the next problem where somewhere else a string is expected instead of an array of URLs. At this point I gave up and just specified the map fully, and then the preview was fine. I also suspect this would be the easiest solution: say in the docs that the map must be fully specified and include/reference the example from ckanext-spatial for Stamen Watercolour but use Terrain instead as CKAN itself is trying to do.

**Comment by pduchesne** [comment]
Indeed, there have been significant changes since the ckanext-spatial fork, and merely redirecting to the 'old' ckanext-spatial doc is no longer valid.
I'll rewrite that doc.
